src/runner.ts

```diff
@@ -374,12 +406,31 @@ This is an automated system message and there is no user available to respond. P
     tools: Tool[],
   ): Promise<Result<Message[], SrchdError>> {
     let tokenCount = 0;
+    /*
+     * Conversation Invariants:
```

Comment:
The invariant should be on `this.contextPruning` fields as we discussed and was present in previous commits 🙏